### PR TITLE
feat(indexer): Publish typescript types for api

### PR DESCRIPTION
**Pull request by roninjin10**
- Generate indexer types for typescript projects using indexer
- Added a typesafe function for creating an indexer uri too

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7238** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7238" target="_blank">Graphite</a>  👈
    * **PR #7245** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7245" target="_blank">Graphite</a> 
    * **PR #7221** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7221" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7238?utm_source=stack-comment).

File: indexer/ts/indexer.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export * as indexerTypes from './generated'
```
Comment (roninjin10) [inline indexer/ts/indexer.ts]: ### 
```suggestion
export * from './generated'
```
Typo

File: indexer/ts/indexer.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+  return `?${queries.join('&')}`
+}
+
+export const depositEndpoint = ({ baseUrl = '', address, cursor, limit }: Options): string => {
+  return `${baseUrl}/api/deposits/${address}/${createQueryString({ cursor, limit })}`
+}
+
+export const withdrawalEndoint = ({ address, cursor, limit }: Options): string => {
+  return `/api/withdrawals/${address}/${createQueryString({ cursor, limit })}`
```
Comment (roninjin10) [inline indexer/ts/indexer.ts]:
```suggestion
  return `${baseUrl}/withdrawals/${address}/${createQueryString({ cursor, limit })}`
```

File: indexer/ts/indexer.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    queries.push(`limit=${limit}`)
+  }
+  return `?${queries.join('&')}`
+}
+
+export const depositEndpoint = ({ baseUrl = '', address, cursor, limit }: Options): string => {
+  return `${baseUrl}/api/deposits/${address}/${createQueryString({ cursor, limit })}`
+}
+
+export const withdrawalEndoint = ({ address, cursor, limit }: Options): string => {
```
Comment (roninjin10) [inline indexer/ts/indexer.ts]:
```suggestion
export const withdrawalEndoint = ({ baseUrl = '', address, cursor, limit }: Options): string => {
```

File: indexer/ts/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "private": true,
+  "name": "@eth-optimism/indexer",
```
Comment (hamdiallam) [inline indexer/ts/package.json]: should we name this `indexer-api?. Just so it's distinctly about the api side of this project and not the "index" service.

Same thoughy regarding naming the folder `ts-api`.

This is a nit though, ffti

**Review by hamdiallam — APPROVED** [review]

File: indexer/ts/package.json (hunk 4 above)
Comment (roninjin10) [reply indexer/ts/package.json]: indexer-api is better agreed

**Comment by roninjin10** [comment]
Gonna validate the package works with a canary release before merging

**Pull request merged**